shared/src/main/scala/AST.scala

```diff
@@ -56,7 +56,18 @@ enum AST(val arity: Option[Int]):
     case Str(value)      => s"\"$value\""
     case Lst(elems)      => elems.map(_.toVyxal).mkString("#[", "|", "#]")
     case Command(value)  => value
-    case Group(elems, _) => elems.map(_.toVyxal).mkString
+    case Group(elems, _) =>
+      // replace instances of Number, Number with Number, Space, Number
+      elems
+        .groupBy(_ match
+          case Number(_) => true
+          case _         => false
+        )
```

Comment:
```suggestion
        .groupBy {
          case Number(_) => true
          case _         => false
        }
```
Or just `.groupBy(_.isInstanceOf[AST.Number])`